pkg/cloudprovider/gce/gce.go

```diff
@@ -421,6 +464,57 @@ func (gce *GCECloud) GetNodeResources(name string) (*api.NodeResources, error) {
 	}
 }
 
+func getMetadataValue(metadata *compute.Metadata, key string) string {
+	for _, item := range metadata.Items {
+		if item.Key == key {
+			return item.Value
+		}
+	}
+	return ""
+}
+
+func (gce *GCECloud) Configure(name string, spec *api.NodeSpec) error {
+	instanceName := canonicalizeInstanceName(name)
+	instanceCall := gce.service.Instances.Get(gce.projectID, gce.zone, instanceName)
+	instance, err := instanceCall.Do()
+	if err != nil {
+		return err
+	}
+	if currentValue := getMetadataValue(instance.Metadata, podCIDRMetadataKey); currentValue != "" {
+		if currentValue == spec.PodCIDR {
+			// IP range already set to proper value.
+			return nil
+		}
+		return ErrMetadataConflict
+	}
+	instance.Metadata.Items = append(instance.Metadata.Items,
+		&compute.MetadataItems{
+			Key:   podCIDRMetadataKey,
+			Value: spec.PodCIDR,
+		})
+	setMetadataCall := gce.service.Instances.SetMetadata(gce.projectID, gce.zone, instanceName, instance.Metadata)
+	setMetadataOp, err := setMetadataCall.Do()
+	if err != nil {
+		return err
+	}
+	err = gce.waitForZoneOp(setMetadataOp)
+	if err != nil {
+		return err
+	}
+	insertCall := gce.service.Routes.Insert(gce.projectID, &compute.Route{
+		Name:            instanceName,
+		DestRange:       spec.PodCIDR,
+		NextHopInstance: fmt.Sprintf("zones/%s/instances/%s", gce.zone, instanceName),
+		Network:         fmt.Sprintf("global/networks/%s", gce.networkName),
+		Priority:        1000,
+	})
+	insertOp, err := insertCall.Do()
+	if err != nil {
+		return err
+	}
+	return gce.waitForGlobalOp(insertOp)
```

Comment:
Along with creating the routes in parallel, we also have to think about who tears down the routes (and how).

I don't really like the asymmetric route creation/deletion (i.e. nodecontroller.go creates routes, but util.sh cleans them up). But I'm not particularly excited about making the master responsible for deleting them either. Right now, GKE only cleans up what it created, so we'll have to look at how we're going to handle this.